packages/i18n/I18nDecorator/getI18nClasses.js

```diff
@@ -1,60 +1,93 @@
 		// regular fonts, creating a strange 'ransom note' look with a mix of fonts in the
 		// same word. So, treat it like a non-Latin language in order to get all the characters
 		// to display with the same font.
-		classes.push(base + 'non-latin');
-	}
-
-	const scriptName = li.getScript();
-	if (scriptName !== 'Latn' && scriptName !== 'Cyrl' && scriptName !== 'Grek') {
-		// GF-45884: allow enact to avoid setting italic fonts for those scripts that do not
-		// commonly use italics
-		classes.push(base + 'non-italic');
-	}
-
-	// allow enact to apply right-to-left styles to the app and widgets if necessary
-	if (isRtlLocale()) {
-		classes.push(base + 'right-to-left');
-	}
-
-	// allow enact or the apps to give CSS classes that are specific to the language, country, or script
-	if (locale.getLanguage()) {
-		classes.push(base + locale.getLanguage());
-		if (locale.getScript()) {
-			classes.push(base + locale.getLanguage() + '-' + locale.getScript());
-			if (locale.getRegion()) {
-				classes.push(base + locale.getLanguage() + '-' + locale.getScript() + '-' + locale.getRegion());
+		(done) => isNonLatinLocale(locale, {
+			...options,
+			onLoad: (isNonLatin) => done(isNonLatin ? base + 'non-latin' : '')
+		}),
+		// allow enact to apply right-to-left styles to the app and widgets if necessary
+		(done) => isRtlLocale({
+			...options,
+			onLoad: (isRtl) => done(isRtl ? base + 'right-to-left' : '')
+		})
+	], function (classes) {
+		const scriptName = li.getScript();
+		if (scriptName !== 'Latn' && scriptName !== 'Cyrl' && scriptName !== 'Grek') {
+			// GF-45884: allow enact to avoid setting italic fonts for those scripts that do not
+			// commonly use italics
+			classes.push(base + 'non-italic');
+		}
+
+		// allow enact or the apps to give CSS classes that are specific to the language, country, or script
+		if (locale.getLanguage()) {
+			classes.push(base + locale.getLanguage());
+			if (locale.getScript()) {
+				classes.push(base + locale.getLanguage() + '-' + locale.getScript());
+				if (locale.getRegion()) {
+					classes.push(base + locale.getLanguage() + '-' + locale.getScript() + '-' + locale.getRegion());
+				}
+			} else if (locale.getRegion()) {
+				classes.push(base + locale.getLanguage() + '-' + locale.getRegion());
 			}
-		} else if (locale.getRegion()) {
-			classes.push(base + locale.getLanguage() + '-' + locale.getRegion());
 		}
-	}
-	if (locale.getScript()) {
-		classes.push(base + locale.getScript());
-	}
-	if (locale.getRegion()) {
-		classes.push(base + locale.getRegion());
-	}
-
-	return classes.join(' ');
+		if (locale.getScript()) {
+			classes.push(base + locale.getScript());
+		}
+		if (locale.getRegion()) {
+			classes.push(base + locale.getRegion());
+		}
+
+		options.onLoad(classes.filter(Boolean).join(' '));
+	});
+}
+
+/*
+ * A function that returns locale in class name.
+ *
+ * @memberof i18n/I18nDecorator
+ * @returns {String} Class
```

Comment:
This needs an update

Reply:
Well, it _eventually_ returns a string ... sort of ...

... Okay, I'll update it!